### PR TITLE
Fy double declining balance

### PR DESCRIPTION
**Pull request by theyeeman**

### CODE REVIEW

File: pfinance/functions.py [hunk 1]
```diff
@@ -257,6 +257,49 @@ def straight_line_depreciation(purchase_price: float, salvage_value: float, usef
     return (purchase_price - salvage_value) / useful_life
 
 
+def double_declining_balance_depreciation(
+    purchase_price: float,
+    salvage_value: float,
+    useful_life: int,
+    factor: float = 2.0,
```
Comment (j-pettit) [inline pfinance/functions.py:264]: Is there a reason for making a declining balance a positive value? Might be more clear to use 0.5 for the same effect

File: pfinance/functions.py [hunk 2]
```diff
@@ -257,6 +257,49 @@ def straight_line_depreciation(purchase_price: float, salvage_value: float, usef
...
+) -> dict[list[float], list[float]]:
+    '''
+    Calculate the depreciation of an asset using double declining balance.
+
+        Parameters:
+            purchase_price (float): The total amount paid for the asset
+            salvage_value (float): The value of the asset after its useful life
+            uselife_life (int): The expected lifespan of an asset, must be greater than 0
```
Comment (j-pettit) [inline pfinance/functions.py:272]: should be `useful_life`

File: pfinance/functions.py [hunk 3]
```diff
@@ -257,6 +257,49 @@ def straight_line_depreciation(purchase_price: float, salvage_value: float, usef
...
+            factor (int): The rate at which the balance declines, default 2
+
+        Returns:
+            double_declining_balance_result (dict):
+                asset_value (list[float]): Value of the asset at beginning of the period
+                periodic_depreciation (list[float]): Deprecriation of the asset at end of the period
+    '''
+    asset_value = [purchase_price * 1.0]
```
Comment (j-pettit) [inline pfinance/functions.py:280]: What's the purpose of the `* 1.0`?

File: pfinance/functions.py [hunk 4]
```diff
@@ -257,6 +257,49 @@ def straight_line_depreciation(purchase_price: float, salvage_value: float, usef
     return (purchase_price - salvage_value) / useful_life
 
 
+def double_declining_balance_depreciation(
+    purchase_price: float,
+    salvage_value: float,
+    useful_life: int,
+    factor: float = 2.0,
+) -> dict[list[float], list[float]]:
```
Comment (j-pettit) [inline pfinance/functions.py:265]: I believe this should be `dict[string, list[float]]` (annotation shows the key type and value type)

File: pfinance/functions.py (hunk 1 above)
Comment (theyeeman) [reply pfinance/functions.py:264]: I want to follow the format that Excel uses

File: pfinance/functions.py (hunk 3 above)
Comment (theyeeman) [reply pfinance/functions.py:280]: I was thinking of keeping all the values inside asset_value as a float which is why I multiplied by 1.0 (but now that I think about it, i could just cast to float instead).

Comment (j-pettit) [reply pfinance/functions.py]: Either is fine with me, even if you just decide it doesn't need to be a float that would work too (looks better as a float for display purposes but any division, etc would treat an int as a float anyways)

**Review by j-pettit — APPROVED** [review]

**Pull request merged**